### PR TITLE
Javadocs, rendering & testmod item

### PR DESCRIPTION
**Pull request by florensie**
- Update and add javadocs
- Rendering with LivingEntities and EntityModels
- followBodyRotations [from Curios](https://github.com/TheIllusiveC4/Curios/blob/d107f06b020a4f94e9e6dad374daf74d1f8e0de3/src/main/java/top/theillusivec4/curios/api/type/component/IRenderableCurio.java#L149) but slightly modified to only take one `BipedEntityModel`
This is basically an all in one for the `translateTo*` methods, should those be kept?
- Fix some possible bugs
- Add working trinket to test mod

### CODE REVIEW

File: src/testmod/java/dev/emi/trinkets/TestTrinket.java
Comment (emilyalexandra) [inline src/testmod/java/dev/emi/trinkets/TestTrinket.java:22]: This UUID behavior may be a holdover from experience you may have in curios, but that's actually a behavior trinkets corrects, you should be constructing an `EntityAttributeModifier` with the passed UUID.

Explanation if you want to know:
Using a constant UUID means that your effect won't stack, and only apply once, worse, if you wear 2 of the same trinket and then remove one, you will not get the effect at all (this is because unequiping removes the EAM and minecraft tracks EAMs using UUIDs). There are also some other edge cases that will result in unintended behavior. This is the sort of thing you don't have to worry about with armor slots or held items because only a single instance can be applied at once.

**Review by emilyalexandra — CHANGES_REQUESTED** [review]
Other than the EAM UUID issue, this is good. Particularly, me and C4 were talking about fixing the renderer being player entity specific. I think it's useful to keep the translate methods, as different approaches without models (like rendering items or blocks) would probably appreciate it.

File: src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java
Comment (TheIllusiveC4) [inline src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java:41]: Is there a particular reason that this was modified from Curios to take only one model?

File: src/testmod/java/dev/emi/trinkets/TestTrinket.java
Comment (florensie) [reply src/testmod/java/dev/emi/trinkets/TestTrinket.java:22]: Oh, that makes a lot of sense, thanks for the explanation.

File: src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java
Comment (florensie) [reply src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java:41]: Correct me if I'm wrong but when you create a BipedModel this way for your trinket I don't expect anyone to supply more than one model, ever. I just don't really see any use for it and maybe makes the api a bit less confusing and it's implementation a bit more readable.

**Comment by florensie** [comment]
> I think it's useful to keep the translate methods, as different approaches without models (like rendering items or blocks) would probably appreciate it.

@emilyalexandra that's the only reason I kept them in for now, though I do still think they might need some work (if possible) as they'll only properly work with player models because of those hardcoded values.

File: src/testmod/java/dev/emi/trinkets/TestTrinket.java
Comment (florensie) [reply src/testmod/java/dev/emi/trinkets/TestTrinket.java:22]: > This UUID behavior may be a holdover from experience you may have in curios

Also correct on this btw 😉

File: src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java
Comment (TheIllusiveC4) [reply src/main/java/dev/emi/trinkets/api/client/TrinketRenderer.java:41]: That's fair. I was just mainly curious if there was a more major reason for it, I think that's fine as it is. It's just a helper method anyway, so it doesn't necessarily need to cover every use-case.

File: src/testmod/java/dev/emi/trinkets/TestTrinket.java
Comment (TheIllusiveC4) [reply src/testmod/java/dev/emi/trinkets/TestTrinket.java:20]: I think this is less a holdover from any Curios experience (although it certainly doesn't help) and more to do with how modders are accustomed to setting up their EAMs in static initializers since vanilla doesn't need this sort of consideration. While I worry this will happen fairly often, I'm not sure how to make this more clear to modders since the javadocs are quite clear on the usage of the UUID parameter.

**Comment by florensie** [comment]
I should probably add some rendering to the test mod before this gets merged, so maybe leave this open until I get to that.

**Comment by TheIllusiveC4** [comment]
> 
> 
> > I think it's useful to keep the translate methods, as different approaches without models (like rendering items or blocks) would probably appreciate it.
> 
> @emilyalexandra that's the only reason I kept them in for now, though I do still think they might need some work (if possible) as they'll only properly work with player models because of those hardcoded values.

I'm not sure those helper methods are really applicable outside of a player context if we try to abstract it out further since there's no good way to really identify where the face or leg or something of a generic entity would ideally be located.

**Comment by emilyalexandra** [comment]
> 
> 
> I should probably add some rendering to the test mod before this gets merged, so maybe leave this open until I get to that.

Okay I'll wait for that, and we can keep the translate method for now, before we make an official 3.0.0 release though I may think about removing them

File: src/testmod/java/dev/emi/trinkets/client/TrinketsTestClient.java
Comment (emilyalexandra) [inline src/testmod/java/dev/emi/trinkets/client/TrinketsTestClient.java:12]: Is this cast necessary?

**Review by emilyalexandra — APPROVED** [review]
Looks good, I'll wait for your response and merge

File: src/testmod/java/dev/emi/trinkets/client/TrinketsTestClient.java
Comment (emilyalexandra) [reply src/testmod/java/dev/emi/trinkets/client/TrinketsTestClient.java:12]: Aha, I forgot that's how items are typically stored.

**Pull request merged**
